src/java/org/apache/cassandra/db/streaming/CassandraStreamHeader.java

```diff
@@ -168,10 +232,35 @@ public void serialize(CassandraStreamHeader header, DataOutputPlus out, int vers
             header.calculateCompressionInfo();
             CompressionInfo.serializer.serialize(header.compressionInfo, out, version);
             out.writeInt(header.sstableLevel);
+
             SerializationHeader.serializer.serialize(header.version, header.header, out);
+
+            header.tableId.serialize(out);
+            out.writeBoolean(header.fullStream);
+
+            if (header.fullStream)
+            {
+                out.writeInt(header.components.size());
+                for (ComponentInfo info : header.components)
+                    ComponentInfo.serializer.serialize(info, out, version);
+
+                ByteBufferUtil.writeWithShortLength(header.firstKey.getKey(), out);
```

Comment:
It's fine I forgot that the keys are hashes so they will always be fixed size and small.

Reply:
OK, actually Jeff pointed out that we do have partitioners where decorated keys are not hashes. So either we find out what the max length is or just use an int/varint and call it good. We don't send this often so it's fine.